R/extract_dials_parameter.R

```diff
@@ -0,0 +1,118 @@
+#' Return a dials parameter object associated with parameters
+#'
+#' `extract_dials_parameter()` can extract a single `dials` parameter object from
+#' different types of objects (e.g. parameter sets, recipes, etc.).
+#' @param x The results of a call to [parameters()], a recipe, model
+#' specification, or workflow.
+#' @param id A single string for the `id` of the parameter.
+#' @param ... Not currently used.
+#' @return A `dials` parameter object.
+#' @examples
+#' glmn_param <- parameters(lambda = penalty(), mixture())
+#' extract_dials_parameter(glmn_param, "lambda")
+#'
+#' # fails:
+#' # extract_dials_parameter(glmn_param, "penalty")
+#' @export
+extract_dials_parameter <- function(x, id, ...) {
+  UseMethod("extract_dials_parameter")
+}
```

Comment:
I was under the impression that we were going to move `extract_dials_parameter()` to hardhat and only implement the method for `parameters` here

Right now we are in a pretty good spot because dials uses `pull_dials_object()` which has a completely different name than `extract_dials_parameter()`. I worry that if we merge this in, then when it comes time to move the generic to hardhat we could have some issues due to having the generic in both packages (i.e. if a user updates hardhat but not dials somehow)

I imagined that we were going to add `extract_dials_parameter()` to hardhat, then there would be a dials PR that depends on dev hardhat that soft-deprecates `pull_dials_object()` and just implements the method for parameters

Reply:
ah, I hadn't though of that. my goal was to get a dials release done without depending a release of hardhat, recipes, parsnip, and workflows. but I think this is a good reason not to merge this as is. are you planning to do a release of hardhat anytime soon? then I could put the generic there and keep the methods in dials for this release. Or should we leave `pull_dials_object()` as is for this dials release and do all of it in the next round?

Reply:
I vote for leaving `pull_dials_object()` as is since you want to release soon. Then we can do a coordinated release of all the packages next time

Reply:
sounds good :+1: